src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -118,18 +127,27 @@ public static Tag parseTag(String tag) throws ParseException {
     }
 
     /**
-     * Parse a string representing amount into an amount.
+     * Parses a string representing amount into an amount.
      */
     public static Amount parseAmount(String amount) throws ParseException {
         requireNonNull(amount);
         String trimmedAmount = amount.trim();
-        if (!StringUtil.isNonZeroUnsignedInteger(trimmedAmount) && !trimmedAmount.equals("0")) {
+        if (!StringUtil.isUnsignedInteger(trimmedAmount)) {
             throw new ParseException(MESSAGE_INVALID_AMOUNT);
```

Comment:
If the input was "9999999999999999999", it will cause a NumberFormatException as `Integer.parseInt()` is not able to parse such a number. So, perhaps you should catch if limit has been exceeded first?

Reply:
I tried and indeed it is not able to parse, will change now, thanks !